idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3999 @@
+                    cst(self.energy_balance_solid_eqn[t,ix,iz],sqx*sLy*sLz)
+                    
+                    for j in self.comps:
+                        sy = sgsf(self.mole_frac_comp[t,ix,iz,j],sy_def)
+                        cst(self.conc_eqn[t,ix,iz,j],sy*sP)
+                        ssf(self.Dconc[t,ix,iz,j],sxflux[j]*sLx/sD)
+                        
+                        cst(self.material_balance_eqn[t,ix,iz,j],
+                            sxflux[j]*sLy*sLz)
+                        
+               
+@declare_process_block_class("SocTriplePhaseBoundary")
+class SocTriplePhaseBoundaryData(UnitModelBlockData):
+    CONFIG = ConfigBlock()
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([useDefault, True, False]),
+            default=useDefault,
+            description="Dynamic model flag",
+            doc="""Indicates whether this model will be dynamic,
+            **default** = useDefault.
+            **Valid values:** {
+            **useDefault** - get flag from parent (default = False),
+            **True** - set as a dynamic model,
+            **False** - set as a steady-state model.}""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(domain=In([False]), default=False),
+    )
+    CONFIG.declare(
+        "comp_list",
+        ConfigValue(default=["H2", "H2O"], description="List of components"),
+    )
+    CONFIG.declare(
+        "tpb_stoich_dict",
+        ConfigValue(
+            description="Stochiometry coefficients for component reactions on "
+            "the triple phase boundary.",
+        ),
+    )
+    CONFIG.declare(
+        "conc_ref",
+        ConfigValue(
+            default=None, 
+            description="Variable for the component concentration in bulk channel "
+        ),
+    )
+    CONFIG.declare(
+        "below_electrolyte",
+        ConfigValue(
+            domain=In([True, False]),
+            description="Variable for the component concentration in bulk channel "
+        ),
+    )
+    
+    _submodel_boilerplate_config(CONFIG)
+    _thermal_boundary_conditions_config(CONFIG,thin=True)
+    _material_boundary_conditions_config(CONFIG,thin=True)    
+    
+    def build(self):
+        super().build()
+        
+        # Set up some sets for the space and time indexing
+        dynamic = self.config.dynamic
+        tset = self.flowsheet().config.time
+        # z coordinates for nodes and faces
+        zfaces = self.config.cv_zfaces
+        self.znodes = pyo.Set(
+            initialize=[
+                (zfaces[i] + zfaces[i+1]) / 2.0
+                for i in range(len(zfaces)-1)
+            ]
+        )
+        comps = self.comps = pyo.Set(initialize=self.config.comp_list)
+        self.tpb_stoich = copy.copy(self.config.tpb_stoich_dict)
+        # TODO maybe let user specify inert species directly? Floating point
+        # equalities make me nervous---Doug
```

Comment:
Floating point equalities should make you nervous, but the stochiometric coefficients aren't calculated, so if they are 0 we should enter an integer 0 in the dict.  Even 0.0 is probably okay as long as it's a constant.  Calculating coefficients could get you into trouble here, but that shouldn't be an issue.

Reply:
I'd prefer to future-proof the code if possible. How about assuming that components that do not have a stoichiometric coefficient specified are inert? The problem is that then users could not specify a coefficient of zero, which may lead to confusion.

The best method is probably just adding an `inert_list` to the config.

Complicating matters is attempting to take into account solid-phase species like O^2- ions and vacancies. I don't think we have cause to worry about inert solid-phase species, though, so assuming the inerts are a subset of gas species should be fine.